client/my-sites/themes/theme-options.js

```diff
@@ -57,7 +59,12 @@ function getAllThemeOptions( { translate, isFSEActive } ) {
 				} )
 			);
 
-			return `/checkout/${ slug }/value_bundle?redirect_to=${ redirectTo }`;
+			const { themeTier } = getThemeTier( state, siteId, themeId );
+			const tierMinimumUpsellPlan = THEME_TIERS[ themeTier?.slug ]?.minimumUpsellPlan;
+			const mappedPlan = getPlan( tierMinimumUpsellPlan );
+			const planPathSlug = mappedPlan?.getPathSlug();
+
+			return `/checkout/${ slug }/${ planPathSlug }?redirect_to=${ redirectTo }`;
```

Comment:
They have slightly different logic, so I think it'll be an over-abstraction right now, and it'll be more readable to keep separate. 

The `upgradePlanForBundledThemes` case always redirects to site-setup (with a particular set of get vars) so that the bundled plugins get installed and will always use the `business` plan (as you can't install plugins on lower plans).